### PR TITLE
Add 9 new exceptional tests

### LINKED ISSUE #542: Adding new exceptional behavior tests
##### Issue Overview

Current test suite is not complete enough to cover all the exceptions in the project. I added 9 new tests to help with it.

##### Steps To Reproduce
1. [step 1]
2. [step 2]

##### Additional Information

Anything relevant to help us resolving the problem. For example if you are using `mvn` you can share output for `mvn --version` as well as dependencies in use `mvn dependency:tree`. OS as well as JDK version would be helpful too.

For long outputs such as stacktraces please use HTML5 `<details>`

```
<details>
 <summary>$mvn --version</summary>
 Maven home: /usr/share/maven/latest
 Java version: 1.7.0_79, vendor: Oracle Corporation
 Java home: /usr/java/jdk1.7.0_79/jre
 Default locale: en_US, platform encoding: UTF-8
 OS name: "linux", version: "4.7.7-200.fc24.x86_64", arch: "amd64", family: "unix"
</details>
```

### PR DESCRIPTION
**Pull request by JiyangZhang**
#### Short description of what this resolves:
Add exceptional tests to existing test files. See [issue](https://github.com/arquillian/arquillian-core/issues/542#issue-2186940428)

#### Changes proposed in this pull request:

- 9 new exceptional tests
-
-

Fixes #542

### CODE REVIEW

**Review by starksm64 — APPROVED** [review]
The equivalent additions should be made to the https://github.com/arquillian/arquillian-jakarta repo that has the newer jakarta package namespace.

**Comment by JiyangZhang** [comment]
Hi @starksm64 ,

Thanks, I plan to send more tests in that repo in the next few days.

File: config/impl-base/src/test/java/org/jboss/arquillian/config/impl/extension/PropertiesParserTestCase.java [hunk 1]
```diff
@@ -468,6 +468,16 @@ public String get() {
         });
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowExceptionIfDescriptorIsNull() {
+	new PropertiesParser().addProperties(null, System.getProperties());
```
Comment (rhusar) [inline config/impl-base/src/test/java/org/jboss/arquillian/config/impl/extension/PropertiesParserTestCase.java:473]: The indentation is completly off in all the changes FYI, please fix this next time. Thanks!

Comment (rhusar) [reply config/impl-base/src/test/java/org/jboss/arquillian/config/impl/extension/PropertiesParserTestCase.java:473]: Ah, even worse, these are tabs, but unfortunately, this project uses spaces...

Comment (jasondlee) [reply config/impl-base/src/test/java/org/jboss/arquillian/config/impl/extension/PropertiesParserTestCase.java:473]: > Ah, even worse, these are tabs, but unfortunately, this project uses spaces...

Nothing unfortunate about that. It's what Charles Babbage would have wanted... :P

**Pull request merged**